tests/NuGet.Services.AzureSearch.FunctionalTests/BasicTests/V2SearchProtocolTests.cs

```diff
@@ -0,0 +1,266 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using BasicSearchTests.FunctionalTests.Core;
+using BasicSearchTests.FunctionalTests.Core.TestSupport;
+using NuGet.Versioning;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace NuGet.Services.AzureSearch.FunctionalTests
+{
+    public class V2SearchProtocolTests : NuGetSearchFunctionalTestBase
+    {
+        public V2SearchProtocolTests(CommonFixture fixture, ITestOutputHelper testOutputHelper)
+            : base(fixture, testOutputHelper)
+        {
+        }
+
+        [Fact]
+        public async Task CanGetEmptyResult()
+        {
+            // Act
+            var result = await V2SearchAsync(new V2SearchBuilder { Query = Constants.NonExistentSearchString });
+
+            // Assert
+            Assert.Equal(0, result.TotalHits);
+            Assert.Empty(result.Data);
+        }
+
+        [Fact]
+        public async Task ShouldGetResultsForEmptyString()
+        {
+            // Act
+            var result = await V2SearchAsync(new V2SearchBuilder { Query = "" });
+
+            // Assert
+            Assert.True(result.TotalHits.HasValue && result.TotalHits.Value > 0, "No results found, should find atleast some results for empty string query.");
+            Assert.NotNull(result.Data);
+        }
+
+        [Theory]
+        [MemberData(nameof(TakeResults))]
+        public async Task TakeReturnsExactResults(int take)
+        {
+            var results = await V2SearchAsync(new V2SearchBuilder { Query = "json", Take = take });
+
+            Assert.NotNull(results);
+            Assert.True(results.TotalHits > take);
+            Assert.True(results.Data.Count == take);
+        }
+
+        [Fact]
+        public async Task SkipDoesSkipThePackagesInResult()
+        {
+            var searchTerm = "json";
+            var skip = 5;
+            var resultsWithoutSkip = await V2SearchAsync(new V2SearchBuilder { Query = searchTerm, Take = 10 });
+            var resultsWithSkip = await V2SearchAsync(new V2SearchBuilder { Query = searchTerm, Skip = skip, Take = 10 - skip });
+
+            Assert.NotNull(resultsWithoutSkip);
+            Assert.NotNull(resultsWithSkip);
+            Assert.True(resultsWithoutSkip.Data.Count > 1);
+            Assert.True(resultsWithSkip.Data.Count > 1);
+            Assert.True(resultsWithoutSkip.Data.Count == resultsWithSkip.Data.Count + skip);
+            var packageIDListThatShouldBeSkipped = resultsWithoutSkip.Data
+                .Select(x => x.PackageRegistration.Id)
+                .Take(skip);
+            var skippedResultPackageIdList = resultsWithSkip.Data
```

Comment:
Nit: I would remove the `skippedResultPackageIdList` variable and do the select as you're creating `commonResults`. I don't feel strongly about this, feel free to ignore